build/dsc/resource.rb

```diff
@@ -82,7 +82,25 @@ def ensurable?
 
     def absentable?
       properties.detect do |p|
-        p.is_ensure? && p.values.any? { |v| v.casecmp('absent') == 0 }
+        p.is_ensure? && p.values.any? { |v| v.casecmp('absent') == 0 || v.casecmp('disable') == 0 }
+      end
+    end
+
+     def absent_value
+      properties.detect do |p|
+        p.is_ensure? && p.values.each do |v|
```

Comment:
I think I would rewrite this part to future proof the schema parsing / prioritize `absent` (in the event `absent` and `disable` are both in the map)

```ruby
def absent_value
  properties.detect do |p|
    return 'absent' if p.is_ensure? && p.values.any { |v| v.casecmp('absent') == 0 }
    return 'disable' if p.is_ensure? && p.values.any { |v| v.casecmp('disable') == 0 }
  end
  throw 'Error processing MOF schema - could not determine valid ensure values'
end
```

Thoughts?